homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -60,12 +57,12 @@ class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     @property
     def name(self):
         """Return the name of the device."""
-        return "Yale Smart Alarm"
+        return self.coordinator.entry.data[CONF_NAME]
 
     @property
     def unique_id(self) -> str:
         """Return the unique ID for this entity."""
-        return "yale_smart_alarm"
+        return self.coordinator.entry.data[CONF_USERNAME]
```

Comment:
I see why you'd like to do this. I'm not a maintainer, but this is still unacceptable according to https://developers.home-assistant.io/docs/entity_registry_index/#unacceptable-sources-for-a-unique-id

Reply:
I did screen the list you first attached (url) and there it says an e-mail can be used as secondary if something is not available (which it's not in this case). Then now on this second url it clearly says an e-mail is not to be used so completely contradictory from first url.
I can use the config entry id (second url says) but that doesn't seem like a good idea either. Any suggestions?

Reply:
@gjohansson-ST you are perfectly right - the wiki is contradictory.

The former one looks most updated to me:
https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#sometimes-acceptable-sources-for-a-unique-id-for-cloud-services 

So the only thing you might want to do is to lower case the username if the user name is an email address, or if the user names are case insensitive.

Reply:
Oh - re-reading one more time. Username/email is acceptable as unique id for _config entries_ for cloud based devices, which is the first link. The second link is for _entities_, which is what we have here. I now see I commented using the wrong link in my original comment - apologies for that.

Is there any information returned from the API that can be used to create a unique id for the _entity_?